### DISCUSSION TITLE
Features are needed as plugable in typescript ( version8)

### DISCUSSION THREAD

**Discussion by Keerthi3021** [TanStack/table#3683] [Q&A] [answered]
In old version(v7), all feature are as plugable. But in typescript why plugin system is removed ? . In our case, we need all features as plugable in typescript. Is there is any way to use features as plugable ?

**Comment by tannerlinsley** [comment, marked as answer]
The plugin system has been removed in v8 for a lot of reasons, a few:
- Cannot be typed well
- Composition is usually better in most cases provided the API has sufficient inversion of control
- Created unreliable compatibility issues between plugins that know nothing about each other

In v8, the core ships will all of the main features built-in, but with the ability to override their pieces of functionality with inversion of control points like `sortRowsFn` (which is responsible for implementing client-side sorting logic).

Sorry for this change, but it was necessary. I'd be happy to help you convert your plugins over to use the new composition patterns in v8 as soon as it's in a beta state.

**Comment by Keerthi3021** [comment]
How to include a new plugin in v8 ?

**Reply by tannerlinsley** [reply]
There are no plugins per-se, but the new way of extending React Table functionality will be to wrap `useTable` with your own function that decorates/changes the options and return value from it.
